lib/autoproj/reporter.rb

```diff
@@ -49,8 +49,28 @@ def error(error)
                 Autoproj.message line, STDERR
             end
         end
+
+        def reset_timer
+            @timer_start = Time.now
+        end
+
+        def elapsed_time
+            return unless @timer_start
+            secs = Time.now - @timer_start
+            return if secs < 1
+
+            [[60, 'sec'], [60, 'min'], [24, 'hour'], [1000, 'day']].map do |count, name|
+                if secs > 0
+                    secs, n = secs.divmod(count)
+                    next if (val = n.to_i) == 0
+                    "#{val} #{val > 1 ? name + 's' : name}"
+                end
+            end.compact.reverse.join(' ')
+        end
+
         def success
-            Autoproj.message("Command finished successfully at #{Time.now}", :bold, :green)
+            elapsed_string = elapsed_time ? " (took #{elapsed_time})" : ''
```

Comment:
nil.to_s is the emptry string. (and ?: is very much un-rubyish). A better way (as in more rubocopyish) would be to just assign ... if 

~~~
elapsed_string = " (took #{elapsed_time})" if elapsed_time
~~~

Reply:
(no need to change, just for future reference)

Reply:
Hmm.. This is new to me.

```ruby
if false
    foo = 'bar'
end
```

I thought `foo` would be unset, in which case `puts foo` would raise `NoMethodError` but Ruby conveniently sets `foo` to `nil` even though the expression is false. Pretty neat :+1:

Reply:
In Ruby, local variables are defined after their first assignation, i.e. they are defined (and initialized to `nil`) in *all lines of the same scope* as soon as the parser sees them assigned.

The rule sounds really weird, but it actually makes a lot of sense in the code.

Reply:
> makes a lot of sense in the code.

Definitely. Thanks!